services/user.py

```diff
@@ -0,0 +1,52 @@
+from django.contrib.auth import get_user_model
+
+
+User = get_user_model()
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None,
+) -> User:
+    user = User.objects.create_user(username=username)
+    user.set_password(password)
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.save()
```

Comment:
The `create_user` method actually sets the password for the user, so there's no need to call `set_password` separately. You can pass the password as an argument to `create_user`. Also, you can pass other optional parameters like `email`, `first_name`, `last_name` directly to `create_user` if they are not `None`. This way, you can avoid multiple `if` checks and call `save()` only once.